### PR TITLE
Add parallel version of edge_betweenness_centrality

### PR DESCRIPTION
**Pull request by jkrajniak**
This adds parallel setup for https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.centrality.edge_betweenness_centrality.html

### CODE REVIEW

**Review by Schefflera-Arboricola — COMMENTED** [review]
Thanks @jkrajniak! 

I've left a few comments below. LMK what u think. 
And there are a few linting errors in this PR which you will be able to see once a maintainer approves running the wf test. you can run `black .` to get rid of most of them.

File: benchmarks/benchmarks/bench_centrality.py
Comment (jkrajniak) [reply benchmarks/benchmarks/bench_centrality.py:20]: Agree, makes more sense, as here we deal with edges.

File: nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py
Comment (jkrajniak) [reply nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py]: I've added random chunking over the available nodes.

File: nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py
Comment (Schefflera-Arboricola) [inline nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py:52]: no need to add this test.

In PR https://github.com/networkx/nx-parallel/pull/63 I've added a single test that collects all functions with `get_chunks` kwarg and runs smoke tests for all of them with a random chunking passed into the `get_chunks` arg. (see [test_get_chunks.py](https://github.com/networkx/nx-parallel/blob/06916f9c14cd1c338319a0e9d305a1b2d75fe93c/nx_parallel/tests/test_get_chunks.py)). 

And if PR https://github.com/networkx/nx-parallel/pull/63 gets merged before this PR then please ensure to put `edge_betweenness_centrality` in the `chk_dict_vals` list in the `test_get_chunks.py` file. Otherwise, I'll make sure it's in the PR https://github.com/networkx/nx-parallel/pull/63.

File: timing/timing_all_functions.py
Comment (Schefflera-Arboricola) [inline timing/timing_all_functions.py:1]: I think you can leave this file unchanged for now as it was initially written to create [this heatmap](https://github.com/networkx/nx-parallel/blob/main/timing/heatmap_all_functions.png) and now the package has a lot more algorithms. So, this file needs to be updated (ref PR https://github.com/networkx/nx-parallel/pull/61).

File: nx_parallel/algorithms/centrality/betweenness.py
Comment (Schefflera-Arboricola) [inline nx_parallel/algorithms/centrality/betweenness.py:107]: In PR https://github.com/networkx/nx-parallel/pull/63, I've updated the docstring format a bit to match the sphinx guidelines. So, you would need to move this networkx link above the `Parameters` section, like this:
```.py
    """The parallel computation is implemented by dividing the nodes into chunks and
        computing edge betweenness centrality for each chunk concurrently.

    networkx.edge_betweenness_centrality : https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.centrality.edge_betweenness_centrality.html

    Parameters
    ------------
    get_chunks : str, function (default = "chunks")
        A function that takes in a list of all the nodes as input and returns an
        iterable `node_chunks`. The default chunking is done by slicing the
        `nodes` into `n` chunks, where `n` is the number of CPU cores.
    """
```

**Review by Schefflera-Arboricola — COMMENTED** [review]
Thanks for the changes @jkrajniak!

I've added a few comments below. Thanks!

File: nx_parallel/algorithms/centrality/betweenness.py
Comment (jkrajniak) [reply nx_parallel/algorithms/centrality/betweenness.py]: thanks, fixed

**Comment by dschult** [comment]
Looks like all tests pass now. @Schefflera-Arboricola can you check your comments as "resolved" -- or leave more comments, or another review. When it's ready, select the "Approve" button while leaving a review.
Thanks for this PR @jkrajniak

File: nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py
Comment (Schefflera-Arboricola) [inline nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py:7]: you can put this `get_chunk` back in the `test_betweenness_centrality_get_chunks` as it is not used anywhere else.

**Comment by Schefflera-Arboricola** [comment]
> can you check your comments as "resolved"

no, I think only the author or maintainers can do that.

File: nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py
Comment (jkrajniak) [reply nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py]: done

**Comment by jkrajniak** [comment]
> > can you check your comments as "resolved"
> 
> no, I think only the author or maintainers can do that.

You can just click on the "Resolve conversation" button for each of the comments.

**Comment by dschult** [comment]
Unless there is no button that says "Resolve conversation".  :}  I think github only allows resolving conversations for maintainers and the original poster.  We are working on getting maintainer status for @Schefflera-Arboricola -- proper channels and all that. Meanwhile I resolved those that I think have been addressed. I wasn't sure about one where `ruff` made changes so it should be looked at.  The `ruff` changes are usually "linting" changes, e.g. formatting, whitespace and indentation/line wraps changes.  So it's probably OK.

File: nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py
Comment (Schefflera-Arboricola) [inline nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py:7]: I don't think there is any need for 2 functions here(i.e. `get_chunk_function` and `get_chunk`) and `ebc` doesn't need to be passed as it is already in the outer function's namespace. LMK what you think. Thanks!

**Comment by Schefflera-Arboricola** [comment]
> Meanwhile I resolved those that I think have been addressed. 

Thanks!

> I wasn't sure about one where `ruff` made changes so it should be looked at. The `ruff` changes are usually "linting" changes, e.g. formatting, whitespace and indentation/line wraps changes. So it's probably OK.

There were no style changes(or any changes) in that file. I'm not sure what @jkrajniak was referring to there.

**Comment by dschult** [comment]
I am assuming that we don't get to see the changes. They were probably made by pre-commit locally before pushed to the repo -- so we would not see them. It all looks good from my perspective too, so I will resolve that comment. Thanks!

File: README.md
Comment (Schefflera-Arboricola) [inline README.md:16]: These links also need to be updated! :)

File: nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py
Comment (Schefflera-Arboricola) [reply nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py]: > And if PR https://github.com/networkx/nx-parallel/pull/63 gets merged before this PR then please ensure to put edge_betweenness_centrality in the chk_dict_vals list in the test_get_chunks.py file. Otherwise, I'll make sure it's in the PR https://github.com/networkx/nx-parallel/pull/63.

PR https://github.com/networkx/nx-parallel/pull/63 is merged now. 

Also, please make sure that you run `sh _nx_parallel/script.sh` to add this algorithm to the `get_info` function.

**Review by Schefflera-Arboricola — APPROVED** [review]
Thank you @jkrajniak :)

**Review by dschult — APPROVED** [review]
Looks good to me!  Thanks!

**Pull request merged**
